### PR TITLE
refactor: convert page components to TypeScript

### PR DESCRIPTION
**Pull request by SychO9**
**Part of #3533**

**Changes proposed in this pull request:**
* Sneaked in an a11y fix. (won't squash merge).
* Converts all JS page components to TS.

**Necessity**

- [x] Has the problem that is being solved here been clearly explained?
- [x] If applicable, have various options for solving this problem been considered?
- [x] For core PRs, does this need to be in core, or could it be in an extension?
- [x] Are we willing to maintain this for years / potentially forever?

**Confirmed**

- [x] Frontend changes: tested on a local Flarum installation.
- [ ] Backend changes: tests are green (run `composer test`).
- [x] Core developer confirmed locally this works as intended.
- [x] Tests have been added, or are not appropriate here.

### CODE REVIEW

File: framework/core/js/src/admin/components/BasicsPage.tsx [hunk 1]
```diff
@@ -2,24 +2,27 @@ import app from '../../admin/app';
 import FieldSet from '../../common/components/FieldSet';
 import ItemList from '../../common/utils/ItemList';
 import AdminPage from './AdminPage';
+import { IPageAttrs } from '../../common/components/Page';
```
Comment (davwheat) [inline framework/core/js/src/admin/components/BasicsPage.tsx]:
```suggestion
import type { IPageAttrs } from '../../common/components/Page';
```

File: framework/core/js/src/admin/components/MailPage.tsx [hunk 2]
```diff
@@ -4,12 +4,32 @@ import Button from '../../common/components/Button';
 import Alert from '../../common/components/Alert';
 import LoadingIndicator from '../../common/components/LoadingIndicator';
 import AdminPage from './AdminPage';
+import { IPageAttrs } from '../../common/components/Page';
```
Comment (davwheat) [inline framework/core/js/src/admin/components/MailPage.tsx]:
```suggestion
import type { IPageAttrs } from '../../common/components/Page';
```

File: framework/core/js/src/admin/components/MailPage.tsx [hunk 3]
```diff
@@ -4,12 +4,32 @@ import Button from '../../common/components/Button';
...
+import Model from '../../common/Model';
+import type { ApiResponseSingle } from '../../common/Store';
```
Comment (davwheat) [inline framework/core/js/src/admin/components/MailPage.tsx]: I don't think these imports are used.

```suggestion
```

**Comment by SychO9** [comment]
@davwheat friendly poke :)

**Review by davwheat — APPROVED** [review]

**Pull request merged**